kv/memberlist/memberlist_client.go

```diff
@@ -1005,6 +1034,34 @@ func (m *KV) notifyWatchersSync(key string) {
 	}
 }
 
+func (m *KV) Delete(key string) error {
+	m.storeMu.Lock()
+
+	val, ok := m.store[key]
+	if !ok || val.Deleted {
```

Comment:
@aldernero , if we return here the mutex is never gonna be unlocked, no ? 🤔 

 ```suggestion
	m.storeMu.Lock()
	val, ok := m.store[key]
	m.storeMu.Unlock()
	    
	if !ok || val.Deleted {
	     return nil
	}
	
```

Reply:
yes, nice catch, I updated the code